modules/jdbc/src/main/java/org/testcontainers/containers/JdbcDatabaseContainer.java

```diff
@@ -123,26 +114,31 @@ public SELF withInitScript(String initScriptPath) {
 
     @Override
     protected void waitUntilContainerStarted() {
-        // Repeatedly try and open a connection to the DB and execute a test query
-
         logger().info("Waiting for database connection to become available at {} using query '{}'", getJdbcUrl(), getTestQueryString());
-        Unreliables.retryUntilSuccess(getStartupTimeoutSeconds(), TimeUnit.SECONDS, () -> {
```

Comment:
The changes in this class are the core of this PR:

I've removed use of duct-tape `Unreliables` as it isn't a great fit for the fail-fast behaviour we want. Also, `Unreliables`' currently can't _properly_ enforce timeout on code that would otherwise infinitely loop (TLDR, it doesn't cancel the Future, which is a bug. I'd like to fix this separately).

I originally dropped in Awaitility as a replacement, but realised this was a mistake when shading it; it has a dependency on Hamcrest which seems to break when shaded, and I generally became quite uncomfortable with having Hamcrest in our dependency tree at all.

Despite not using a library any more, I hope that the code here is actually fairly clear in its intent.